New get started guide for the Collector (#3426)
open-telemetry/opentelemetry.io: content/en/docs/collector/getting-started.md

   ```

   After five seconds, telemetrygen stops and shows the sends messages similar
   to the following to the console:

   ```text
   2023-10-23T12:58:19.835+0200	INFO	traces/worker.go:88	traces generated	{"worker": 0, "traces": 994418}
   2023-10-23T12:58:19.835+0200	INFO	traces/traces.go:79	stop the batch span processor
   ```

   The Collector container logs, on the other hand, should show activity related
   to the logs ingest similar to this:

   ```text
   Span #434
      Trace ID       : ba7ef95fce7499811ca72158350c907c
      Parent ID      : 1d3c9f49b3f2cf47
      ID             : 7609079dc6253034
      Name           : okey-dokey
      Kind           : Server
      Start time     : 2023-10-23 11:01:29.53251 +0000 UTC
      End time       : 2023-10-23 11:01:29.532634 +0000 UTC
      Status code    : Unset
      Status message :
   ```

4. Open `http://localhost:55679/debug/tracez` in your browser and select one of
   the samples in the table to see the traces you've just generated.

## Next steps

In this tutorial you've started the OpenTelemetry Collector and sent telemetry
to it. As next steps, consider doing the following:

> - Learn about the different modes of the Collector in
>   [Deployment Methods](../deployment/).

Reviewer: ```suggestion
- Explore different ways to [install the collector](../install/).
- Learn about the different modes of the collector in
  [Deployment Methods](../deployment/).
```

> - Familiarize yourself with the Collector

Reviewer: ```suggestion
- Familiarize yourself with the collector
```

  [configuration](/docs/collector/configuration) files and structure.

> - Try additional components using the
>   [opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib)
>   project.

Reviewer: ```suggestion
- Explore available components in the [registry](https://opentelemetry.io/ecosystem/registry/?language=collector)
- Learn how to [build a custom collector with the OpenTelemetry Collector Builder (OCB)](/docs/collector/custom-collector/)
```

This is an opportunity to promote the registry, I think in that context it is also good to share the existence of OCB, since we want to discourage people from using otel-collector-contrib distro and build their own with the components they need.

---
title: Getting Started
cSpell:ignore: docker dokey dpkg okey telemetrygen
weight: 1
---

> The OpenTelemetry Collector listens to [traces](/docs/concepts/signals/traces/),

Reviewer: ```suggestion
The OpenTelemetry Collector receives [traces](/docs/concepts/signals/traces/),
```

I think that's the more accurate term since we have "receivers"?

Author: +1

[metrics](/docs/concepts/signals/metrics/) and
[logs](/docs/concepts/signals/logs/), processes the telemetry, and exports it to

> a wide variety of observability back-ends using its components. For a conceptual

Reviewer: ```suggestion
a wide variety of observability backends using its components. For a conceptual
```

One more to discuss, we use `backend` more often than `back-end` throughout the docs

Author: Right! FWIW, here's what the Splunk style guide has to say on the matter: https://docs.splunk.com/Documentation/StyleGuide/current/StyleGuide/Usagedictionary#B

OTOH, Google goes the other way: https://developers.google.com/style/word-list#letter-b

Reviewer: Yeah, there is no consistency. I looked up k8s website, they have a preference for `backend` but some instances of `back end` and `back-end` as well, so we don't need to be religious here

   ```

3. The telemetrygen command generates dummy telemetry for testing. Try sending
   some traces to the Collector:

   ```sh
   telemetrygen traces --otlp-insecure --duration 5s
   ```

   After five seconds, `telemetrygen` stops and shows the sended messages in the
   console:

   ```text
   2023-10-23T12:58:19.835+0200	INFO	traces/worker.go:88	traces generated	{"worker": 0, "traces": 994418}
   2023-10-23T12:58:19.835+0200	INFO	traces/traces.go:79	stop the batch span processor
   ```

   In the terminal window, running the Collector container, logs should show
   activity related to the trace ingest similar to this:

   ```text
   Span #434
      Trace ID       : ba7ef95fce7499811ca72158350c907c
      Parent ID      : 1d3c9f49b3f2cf47
      ID             : 7609079dc6253034
      Name           : okey-dokey
      Kind           : Server
      Start time     : 2023-10-23 11:01:29.53251 +0000 UTC
      End time       : 2023-10-23 11:01:29.532634 +0000 UTC
      Status code    : Unset
      Status message :
   ```

4. Open `http://localhost:55679/debug/tracez` in your browser and select one of

>    the samples in the table to see the traces you've just generated.

Reviewer: @theletterf just a suggestion (since we have to wait ;-)): what about adding optional extra steps for metrics and logs as well (if telemetrygen can do them)

Author: Good point. Would tabs be adequate in this case?

Author: Whoops. Status of metrics and logs is still "development". I wouldn't try adding them to the tutorial. +CC @mx-psi

Reviewer: Then let's not add them :-)
